src/NHibernate/Logging.log4net.cs

```diff
@@ -12,14 +13,56 @@ namespace NHibernate
 	public class Log4NetLoggerFactory: ILoggerFactory
 #pragma warning restore 618
 	{
-		private static readonly System.Type LogManagerType = System.Type.GetType("log4net.LogManager, log4net");
+		internal static readonly Assembly Log4NetAssembly;
+		internal static readonly Exception LogManagerTypeLoadException;
+		private static readonly System.Type LogManagerType;
 		private static readonly Func<Assembly, string, object> GetLoggerByNameDelegate;
 		private static readonly Func<System.Type, object> GetLoggerByTypeDelegate;
 
 		static Log4NetLoggerFactory()
 		{
-			GetLoggerByNameDelegate = GetGetLoggerByNameMethodCall();
-			GetLoggerByTypeDelegate = GetGetLoggerMethodCall<System.Type>();
+			try
```

Comment:
It may help diagnosing log4net loading failure. By default they will be silent, but someone having troubles getting log4net logging could try to explicitly enable it with the code `LoggerProvider.SetLoggersFactory(new Log4NetLoggerFactory());` by example, and then the exception will be raised.